### ISSUE TITLE
Branch organization for Coq versions

### ISSUE THREAD

**Issue by JasonGross** [HoTT/HoTT#883]
How do we want to organize branches for different versions of Coq?  In particular, should `master` track trunk, or should it track the latest released version of Coq?  I see four possibilities:
1. `master` tracks Coq trunk; new development happens on `master`, version-specific branches are considered stale/locked
2. `master` tracks Coq trunk; new development happens on either `master` or version-specific branches, and version-specific branches are merged into master periodically
3. `master` tracks Coq trunk; new development happens on either `master` or version-specific branches, and master is merged into version-specific branches periodically
4. `master` tracks the latest released version of Coq, we have a `trunk` branch that tracks trunk, but on which no new development happens; someone (me?) maintains the `trunk` branch either by rebase+force-push or by merging `master` into it; when a new stable version of Coq is released, we branch off the old version from `master` (so, when 8.7 is released, we branch off `v8.6`), and merge `trunk` into `master` and update `master` to track the new version.

I'm partial to either 2 or 4; 2 if we want to support developments on trunk before it's released (e.g., experiments with universe cumulativity, induction-recursion, etc), or 4 if we want to support trunk only for Coq's ci-testing.

**Comment by mikeshulman** [comment]
I don't understand how 3 would work; if `master` has moved on from Coq v8.6, then how could it be merged into a `v8.6` branch?

4 sounds complicated and error-prone.

What is the advantage of 2 over 1?  Having development happen on multiple branches seems also complicated and error-prone.

**Comment by JasonGross** [comment]
3 does not work well, I expect.  It would rely on Coq not changing too much between versions.

4 only works if no-one does development on the trunk branch, and there's only one person who touches it.

2 gives more support for people who don't want to install a Coq trunk.  For example, we could have the "official" branch be `v8.6` until 8.7 is released.  I agree, though, that it is more complicated, and I'd be fine with 1.

**Comment by mikeshulman** [comment]
I could live with 2 if others think it's better, but I'm slightly more inclined to the simpler 1.

**Comment by andrejbauer** [comment]
We do not have manpower to support lots of versions of the library. We have a two-dimensional problem:

* support with respect to Coq versions
* support with respect to development

For development only, a common strategy is to have `stable` (for users) and `master` (for developers). I am not familiar with how people deal with dependence on another piece of software that changes often, but I suppose again it would make sense to have two branches, one fixed to the current official release of Coq, and another one tracking Coq development.

But how do we solve both dimensions at once? Perhaps like this:

* the branch `stable` has no development and works with the latest official Coq release
* the branch `master` has both development and it tracks Coq development at the same time

I think this is like @JasonGross's point 1, except a bit more specific (and simplified).

**Comment by mikeshulman** [comment]
That would be fine with me too.  In fact, on consideration I think I would rather not accumulate *branches* for old versions of Coq and keep them around indefinitely; that seems rather to be the role of tags.  So I suggest augmenting Andrej's proposal with a git tag for the "stable tip" that works with each official Coq release, with the `stable` branch always pointing to the most recent of these.

**Comment by JasonGross** [comment]
I've made a [V8.6 tag](https://github.com/HoTT/HoTT/tree/V8.6), and will soon be merging the v8.7 PR, now that 8.7.0 has been released

**Comment by JasonGross** [comment]
I've updated the stable branch to point to current master; I'll update it again once #851 is merged

**Comment by spitters** [comment]
#851 Has been merged now.

**Comment by ejgallego** [comment]
I guess this is solved, right @JasonGross ?

**Comment by Alizter** [comment]
@JasonGross Can this be closed now?

**Comment by Alizter** [comment]
I'm closing this since we have tags tied to release versions of coq.
